src/Exercise/LandingArea/LandingArea.cs

```
@@ -0,0 +1,149 @@
+        {
+            _areaSize = areaSize;
+            _platformSize = platformSize;
+            _initPlatformPositionX = initPlatformPositionX - 1;
+            _initPlatformPositionY = initPlatformPositionY - 1;
+            InitializeMatrix();
+        }
+
+        private void InitializeMatrix()
+        {
+            //Set landing area
+            var matrix = new int[_areaSize, _areaSize];
+
+            if (_initPlatformPositionX + _platformSize < _areaSize && _initPlatformPositionY + _platformSize < _areaSize)
+            {
+                //Set landing platform
+                for (var i = _initPlatformPositionX; i < _initPlatformPositionX + _platformSize; i++)
+                {
+                    for (var j = _initPlatformPositionY; j < _initPlatformPositionY + _platformSize; j++)
+                    {
+                        matrix.SetValue(1, i, j);
+                    }
+                }
+
+                _matrix = matrix;
+            }
+        }
+
+        /// <summary>
+        /// Checks if a given position is on the platform
+        /// </summary>
+        /// <param name="position">The position to check</param>
+        /// <returns>true if the position is on the platform, false otherwise</returns>
+        public bool IsPositionInPlatform(Position position)
+        {
+            var isPositionInPlatform = false;
+            try
+            {
+                if (position != null && _matrix != null)
+                {
+                    if ((position.X - 1 >= 0 && position.X - 1 < _areaSize) &&
+                        (position.Y - 1 >= 0 && position.Y - 1 < _areaSize))
+                    {
+                        var pos = _matrix.GetValue(position.X - 1, position.Y - 1);
+                        isPositionInPlatform = (bool)pos.Equals(1);
+                    }
+                }
+            }
+            catch
+            {
+                isPositionInPlatform = false;
+            }
```

Adding try/catch hiddes possible errors on the implementation, making very hard to debug issues on production environments. It is best to let exceptions bubble up and arise at the latest layer possible.

---

You are right, At least I should add an error log